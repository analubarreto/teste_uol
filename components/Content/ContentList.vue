<template>
  <el-row :gutter="10">
    <el-col v-for="content in contents" :key="content.id" class="col" :xs="24" :sm="12" :md="8" :lg="6" :xl="6">
      <ContentCard :content="content" />
    </el-col>
  </el-row>
</template>

<script>
// import gql from 'graphql-tag';
import ContentCard from './ContentCard.vue';
export default {
  name: 'ContentList',
  components: {
    ContentCard
  },
  props: {
    contents: {
      type: Array,
      default: () => {
        // eslint-disable-next-line vue/require-valid-default-prop
        return {
          content: []
        }
      }
    }
  }
}
</script>

<style scoped lang="scss">
.col {
  display: flex;
  align-items: center;
  justify-content: center;
  margin-bottom: 2rem;
}
</style>
