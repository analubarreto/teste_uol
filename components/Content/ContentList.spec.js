import { shallowMount } from '@vue/test-utils';
import ContentList from './ContentList.vue';

describe('ContentList', () => {
  test('gets data from the api', async () => {
    const wrapper = shallowMount(ContentList);
    await wrapper.setData({
      contents: [
        {
          id: 'some-id',
          embeddable: true,
          allow_download: false,
          type: 'document',
          updated_at: '1638383143749',
          created_at: '1638383143749',
          title: 'Some title',
        }
      ]
    })
    expect(wrapper.element).toMatchSnapshot()
  });
})
