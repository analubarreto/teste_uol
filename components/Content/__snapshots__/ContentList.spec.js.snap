// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContentList gets data from the api 1`] = `
<el-row
  gutter="10"
>
  <el-col
    class="col"
    lg="6"
    md="8"
    sm="12"
    xl="6"
    xs="24"
  >
    <contentcard-stub
      content="[object Object]"
    />
  </el-col>
</el-row>
`;
